linker/symbols.go

```diff
@@ -43,7 +43,12 @@ type packageSymbols struct {
 	children map[protoreflect.FullName]*packageSymbols
 	files    map[protoreflect.FileDescriptor]struct{}
 	symbols  map[protoreflect.FullName]symbolEntry
-	exts     map[protoreflect.FullName]map[protoreflect.FieldNumber]ast.SourcePos
+	exts     map[extNumber]ast.SourcePos
```

Comment:
This looked like low-hanging fruit -- nested maps are often less dense/more wasteful than a single map. But files typically don't have _enough_ extensions for this to be a material source of memory usage. So this wasn't really a noticeable improvement. I left the change in anyway since it made the insertion logic cleaner and the tests a little less verbose.